### PR TITLE
Print how to reattch to seesion

### LINKED ISSUE #17329: Print how to reattach to a session when it times out.
After a `kubectl run` times out, you can usually reattach with something like  `kubectl attach -i --tty foo-a3e6b`.

But not all users realize that.  It would be nice, when a `kubectl run` or `kubectl attach` session ends, to print `Session ended.  Resume with  ./kubectl attach -i --tty <container-name>`

### PR DESCRIPTION
**Pull request by feihujiang**
closes #17329

### CODE REVIEW

**Comment by janetkuo** [comment]
Can you print this message in `kubectl attach` when the session timeout/ends, too?

File: pkg/kubectl/cmd/run.go [hunk 1]
```diff
@@ -213,7 +213,14 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 		if err != nil {
 			return err
 		}
-		return handleAttachPod(f, client, attachablePod, opts)
+		err = handleAttachPod(f, client, attachablePod, opts)
+		if err != nil {
+			return err
+		}
+		if opts.TTY && restartPolicy == api.RestartPolicyAlways {
+			fmt.Fprintf(cmdOut, "Session ended, resume using '$ kubectl attach %s -i -t' command when the pod is running\n", attachablePod.Name)
```
Comment (janetkuo) [inline pkg/kubectl/cmd/run.go]: Remove the $

Comment (janetkuo) [inline pkg/kubectl/cmd/run.go]: add `-c CONTAINER` too

Comment (feihujiang) [inline pkg/kubectl/cmd/run.go]: The pod that created in run command doesn't have more than one container,  I think -c CONTAINER isn't needed.

Comment (feihujiang) [inline pkg/kubectl/cmd/run.go]: Done remove the $ and add -c CONTAINER . kubectl attach and run use the same message, so add -c CONTAINER finally.

**Comment by feihujiang** [comment]
I think who use kubectl attach should know resume session using  kubectl attach firstly. Oh yes, print the message could give the clue, would not be badder than no message. I added it finally.

**Comment by feihujiang** [comment]
@janetkuo PTAL.

**Comment by janetkuo** [comment]
You edit in run.go is gone. Is that intentional?

**Comment by feihujiang** [comment]
Yes, because run.go also invoke ```func (p *AttachOptions) Run() error``` function.

**Comment by janetkuo** [comment]
LGTM. Thanks!

**Pull request merged**